onnx/helper.py

```diff
@@ -359,8 +359,8 @@ def make_tensor(
     tensor.data_type = data_type
     tensor.name = name
 
-    if data_type == TensorProto.STRING:
-        assert not raw, "Can not use raw_data to store string type"
+    if data_type == TensorProto.STRING and raw:
+        raise ValueError("Can not use raw_data to store string type.")
```

Comment:
Should this be TypeError?

Reply:
Yes, thanks.  Updated to TypeError, and also the other lines you suggested below.